src/units/abilities.cpp

```diff
@@ -448,6 +446,34 @@ bool unit::ability_affects_weapon(const config& cfg, const_attack_ptr weapon, bo
 	if(!cfg.has_child(filter_tag_name)) {
 		return true;
 	}
+	const config& filter_student_name= cfg.child("filter_student");
```

Comment:
maybe you could explain the reason for these lines?

Reply:
@gfgtdf it is for avoid conflict with [filter_weapon] in [filter_student/opponent/etc...], like inleadership

Reply:
maybe you coudl give an example of a case where this `if` fixed behviour?

Reply:
@gfgtdf imagine someone creating leadership but foolishly implementing 
[filter_weapon]
[not]
range = ranged

and 
[filter_student]
[filter_weapon]
range = ranged

without the condition if [filter_student],
we would have a conflict
these conditions give priority to the filters contained in [filter_student / opponent / etc ..] so as not to compromise the operation of the game

Reply:
No please give a complete wml example (wthi closing tags) of the ability, with an exact description how it should have in your opinion. To me this code looks like a simple `[damage][filter_student][filter_weapon]` will disable the check for `[damage][filter_weapon]` which is the not how out filters usually work.